Why-NiJeL Frontend Section

josephbosire opened the pull request:
Fixes the why nijel section of the front end. It seems the html part was commented out.
I made some enhancement to code to make it more simpler and also added the ability to dynamically fetch the menu items for that section.
I also added an angular filter to help slugify urls for better SEO support

How to test?

1. Checkout the branch
 `git checkout origin/whynijel_frontend_section`
2. Run gulp
 `gulp`
3. Navigate to the  why nijel page form the menu section and it should now load the content

josephbosire committed:
Why Nijel Fixes

Signed-off-by: Joseph Bosire <kashboss@gmail.com>

josephbosire committed:
Fixes front end why nijel section

Add controller to dynamicall fetch the contents of the why-nijel menu from the db
Fixes bug where the page html was commented out making it not render

Signed-off-by: Joseph Bosire <kashboss@gmail.com>

teddyotieno commented on app/js/controllers/nav.js:
```diff
@@ -0,0 +1,17 @@
+    ClientDataService.fetchWhyNijelSections()
+        .then((resp) => {
+            $scope.titles = [];
+            resp.data.sections.forEach(function (section) {
```
just a nitpick, using arrow functions as opposed to the normal function definitions, for consistency purposes

josephbosire replied:
Done

teddyotieno commented:
Pulled the branch locally, not sure why but's it's not working for me, a click on the dropdown menu doesn't trigger it's corresponding tab, can only see the url being updated.

josephbosire committed:
Codereview fixes

Signed-off-by: Joseph Bosire <kashboss@gmail.com>

josephbosire commented:
@teddyotieno  Could you try again? Testing it locally seems to be working. Also make sure to clear your browser cache.

teddyotieno commented:
@judithachieng possible if you pull down this branch and test it out, report on whether the why-nijel drop-downs are connected to the tabs, clear out the site data before you test it out
<img width="903" alt="screen shot 2017-11-10 at 6 28 02 pm png" src="https://user-images.githubusercontent.com/12892106/32665441-4b4b6be0-c645-11e7-96f3-86a077df619f.png">

judithachieng commented:
@teddyotieno  I have tested it out locally and the drop downs are not connected to the respective tabs.

josephbosire commented:
@teddyotieno and @judithachieng  Hey guys sorry I had not understood the comment teddy had made before and thought he was saying the page is not loading at all. Totally my fault. Yes that is a bug and the menu don't link to the corresponding tab and will work on a fix for that

josephbosire committed:
Bug fix for linking between menu and tabs

This fixes the bug where the menu for the why-nijel section did not navigate to the appropriate tab

Signed-off-by: Joseph Bosire <kashboss@gmail.com>

josephbosire commented:
@teddyotieno  and @judithachieng The bug that relates to the navigation action has now been fixed. Selecting a menu option in the why-nijel drop down menu will now link to the appropriate tab

teddyotieno commented:
Looks good 👍
